From: Guido Draheim <guidod@gmx.de>
Date: Tue, 13 Mar 2018 00:23:33 +0100
Subject: check rootseek and rootsize to be positive #27
Origin: https://github.com/gdraheim/zziplib/commit/8f48323c181e20b7e527b8be7229d6eb1148ec5f
Bug-Debian-Security: https://security-tracker.debian.org/tracker/CVE-2018-7726
Bug-Debian: https://bugs.debian.org/913165
Bug: https://github.com/gdraheim/zziplib/issues/27
Bug: https://github.com/gdraheim/zziplib/issues/41

---
 zzip/zip.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/zzip/zip.c b/zzip/zip.c
index a5db9d8cf9ce..6be8d7c8f944 100644
--- a/zzip/zip.c
+++ b/zzip/zip.c
@@ -318,6 +318,8 @@ __zzip_fetch_disk_trailer(int fd, zzip_off_t filesize,
                     trailer->zz_rootseek = zzip_disk_trailer_rootseek(orig);
                     trailer->zz_rootsize = zzip_disk_trailer_rootsize(orig);
 #                  endif
+                    if (trailer->zz_rootseek < 0 || trailer->zz_rootsize < 0)
+                       return(ZZIP_CORRUPTED); // forged value
 
                     __fixup_rootseek(offset + tail - mapped, trailer);
 		    /*
@@ -344,6 +346,8 @@ __zzip_fetch_disk_trailer(int fd, zzip_off_t filesize,
                         zzip_disk64_trailer_finalentries(orig);
                     trailer->zz_rootseek = zzip_disk64_trailer_rootseek(orig);
                     trailer->zz_rootsize = zzip_disk64_trailer_rootsize(orig);
+                    if (trailer->zz_rootseek < 0 || trailer->zz_rootsize < 0)
+                       return(ZZIP_CORRUPTED); // forged value
 		    /*
 		     * "extract data from files archived in a single zip file."
 		     * So the file offsets must be within the current ZIP archive!
-- 
2.11.0

