From: Guido Draheim <guidod@gmx.de>
Date: Thu, 15 Mar 2018 23:54:37 +0100
Subject: fix for zz_rootsize #41
Origin: https://github.com/gdraheim/zziplib/commit/feae4da1a5c92100c44ebfcbaaa895959cc0829b
Bug-Debian-Security: https://security-tracker.debian.org/tracker/CVE-2018-7726
Bug-Debian: https://bugs.debian.org/913165
Bug: https://github.com/gdraheim/zziplib/issues/27
Bug: https://github.com/gdraheim/zziplib/issues/41

---
 zzip/zip.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/zzip/zip.c b/zzip/zip.c
index 0d79d5298860..14e2e06615cd 100644
--- a/zzip/zip.c
+++ b/zzip/zip.c
@@ -421,7 +421,7 @@ __zzip_parse_root_directory(int fd,
     zzip_off64_t zz_rootseek = _disk_trailer_rootseek(trailer);
     __correct_rootseek(zz_rootseek, zz_rootsize, trailer);
 
-    if (zz_entries < 0 || zz_rootseek < 0 || zz_rootseek < 0)
+    if (zz_entries < 0 || zz_rootseek < 0 || zz_rootsize < 0)
         return ZZIP_CORRUPTED;
 
     hdr0 = (struct zzip_dir_hdr *) malloc(zz_rootsize);
-- 
2.11.0

