From: Guido Draheim <guidod@gmx.de>
Date: Mon, 5 Feb 2018 13:57:49 +0100
Subject: need to check on endbuf for stored files #15
Origin: https://github.com/gdraheim/zziplib/commit/72ec933663f738d8e166979aa7fd5590b2104a07
Bug-Debian-Security: https://security-tracker.debian.org/tracker/CVE-2018-6540
Bug-Debian: https://bugs.debian.org/923659
Bug: https://github.com/gdraheim/zziplib/issues/15

---

--- a/zzip/mmapped.c
+++ b/zzip/mmapped.c
@@ -551,7 +551,12 @@ zzip_disk_entry_fopen(ZZIP_DISK * disk,
     file->avail = zzip_file_header_usize(header);
 
     if (! file->avail || zzip_file_header_data_stored(header))
-        { file->stored = zzip_file_header_to_data (header); return file; }
+    { 
+         file->stored = zzip_file_header_to_data (header);
+         if (file->stored + file->avail >= disk->endbuf)
+             goto error;
+         return file; 
+    }
 
     file->stored = 0;
     file->zlib.opaque = 0;
@@ -560,11 +565,18 @@ zzip_disk_entry_fopen(ZZIP_DISK * disk,
     file->zlib.avail_in = zzip_file_header_csize(header);
     file->zlib.next_in = zzip_file_header_to_data(header);
 
-    if (! zzip_file_header_data_deflated(header) ||
-        inflateInit2(&file->zlib, -MAX_WBITS) != Z_OK)
-        { free (file); return 0; }
+    if (file->zlib.next_in + file->zlib.avail_in >= disk->endbuf)
+         goto error;
+
+    if (! zzip_file_header_data_deflated(header))
+        goto error;
+    if (inflateInit2(&file->zlib, -MAX_WBITS) != Z_OK)
+        goto error;
 
     return file;
+error:
+    free (file);
+    return 0; 
     ____;
 }
 
@@ -601,6 +613,10 @@ zzip_disk_fread(void *ptr, zzip_size_t s
         size = file->avail;
     if (file->stored)
     {
+        if (file->stored + size >= file->endbuf)
+        {
+            return 0; /* ESPIPE */
+        }
         memcpy(ptr, file->stored, size);
         file->stored += size;
         file->avail -= size;
