From: jmoellers <josef.moellers@suse.com>
Date: Fri, 7 Sep 2018 13:55:35 +0200
Subject: One more free() to avoid memory leak.
Origin: https://github.com/gdraheim/zziplib/commit/0e1dadb05c1473b9df2d7b8f298dab801778ef99
Bug-Debian-Security: https://security-tracker.debian.org/tracker/CVE-2018-16548
Bug-Debian: https://bugs.debian.org/910335
Bug: https://github.com/gdraheim/zziplib/issues/58

---
 zzip/zip.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/zzip/zip.c b/zzip/zip.c
index 51a1a4d93a60..bc6c0800e085 100644
--- a/zzip/zip.c
+++ b/zzip/zip.c
@@ -589,6 +589,8 @@ __zzip_parse_root_directory(int fd,
 	    free(hdr0);
 	}
     }                           /* else zero (sane) entries */
+    else
+        free(hdr0);
 #  ifndef ZZIP_ALLOW_MODULO_ENTRIES
     return (entries != zz_entries) ? ZZIP_CORRUPTED : 0;
 #  else
-- 
2.11.0

